### PR TITLE
Add house to reporting

### PR DESCRIPTION
**Pull request by vonElfvin**

### CODE REVIEW

File: src/main/java/com/hedvig/productPricing/service/service/reports/BDXReportGenerator.java [hunk 1]
```diff
@@ -47,6 +47,7 @@ public BDXReportDTO generateBDXReport(final YearMonth reportingMonth) {
 
     while (!currentUnderwritingYear.isAfter(Year.of(reportingMonth.getYear()))) {
 
+      // Household
```
Comment (palmenhq) [inline src/main/java/com/hedvig/productPricing/service/service/reports/BDXReportGenerator.java]: If we feel the need to "chunk" the sections like this i think rather we should move the chunks into separate methods

File: src/main/java/com/hedvig/productPricing/service/service/reports/MiReportGenerator.java [hunk 2]
```diff
@@ -107,7 +111,31 @@ public MiReportDTO generateMiReport(final YearMonth untilPeriod) {
           }
         ));
 
-      reports.put(currentUnderwritingYear, new UnderwritingYearDTO(household, student));
+      // House
```
Comment (palmenhq) [inline src/main/java/com/hedvig/productPricing/service/service/reports/MiReportGenerator.java]: Same here, i really don't like sectioning code by comments

File: src/main/java/com/hedvig/productPricing/service/service/reports/MiReportGenerator.java [hunk 3]
```diff
@@ -6,6 +6,7 @@
 import com.hedvig.productPricing.service.web.dto.report.UnderwritingYearDTO;
 import com.hedvig.productPricing.service.web.dto.report.WrittenMonthDTO;
 import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
```
Comment (palmenhq) [inline src/main/java/com/hedvig/productPricing/service/service/reports/MiReportGenerator.java]: Imo we should use `package javax.validation.constraints.NotNull` instead

File: src/main/java/com/hedvig/productPricing/service/service/reports/RenewalsMiReportHack.java [hunk 4]
```diff
@@ -91,17 +91,34 @@ public RenewalsMiReportHack(
     final List<ProductEntity> studentProducts = productsInUwyr.stream()
       .filter(product -> product.getHouseType().equals(Product.ProductTypes.STUDENT_BRF) || product.getHouseType().equals(Product.ProductTypes.STUDENT_RENT))
       .collect(toList());
+    final List<ProductEntity> houseProducts = productsInUwyr.stream()
```
Comment (meletios) [inline src/main/java/com/hedvig/productPricing/service/service/reports/RenewalsMiReportHack.java]: Should we name this householdBuildingProducts, in order to be consistent ❓

File: src/main/java/com/hedvig/productPricing/service/service/reports/RenewalsMiReportHack.java [hunk 5]
```diff
@@ -91,17 +91,34 @@ public RenewalsMiReportHack(
...
-    final long householdPolicyCount = productsInUwyr.size() - studentPolicyCount;
+    final long housePolicyCount = houseProducts.size();
+    final long householdPolicyCount = productsInUwyr.size() - studentPolicyCount - housePolicyCount;
 
     MonetaryAmount studentEgwp = calculateAgwpForEverything(studentProducts, shortestActiveToByProductId, untilPeriod);
     MonetaryAmount studentAgwp = calculateAgwpForEverything(studentProducts, shortestActiveToByProductId, UNDERWRITING_YEAR_START.plusYears(2));
 
-    MonetaryAmount householdEgwp = calculateAgwpForEverything(productsInUwyr, shortestActiveToByProductId, untilPeriod).subtract(studentEgwp);
-    MonetaryAmount householdAgwp = calculateAgwpForEverything(productsInUwyr, shortestActiveToByProductId, UNDERWRITING_YEAR_START.plusYears(2)).subtract(studentAgwp);
+    MonetaryAmount houseEgwp = calculateAgwpForEverything(houseProducts, shortestActiveToByProductId, untilPeriod);
```
Comment (meletios) [inline src/main/java/com/hedvig/productPricing/service/service/reports/RenewalsMiReportHack.java]: householdBuildingEgwp householdBuildingAgwp

**Pull request merged**